Yoast/yoast-components
# Fix styling react snippet editor

## Summary

Multiple styling changes to make the snippet editor more closely match the original snippet editor.

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Used a prop to allow for different heights, to create a bigger input field for the meta description.
* Changed button.extends to extends(button). This fixes the switcher style, but I am unsure why. 


## Test instructions

This PR can be tested by following these steps:

* Install deps: `yarn install`.
* Link the yoast-components branch:
  * In the yoast-components clone: `yarn link` (skip this step if already done before).
  * In wordpress-seo clone: `yarn link yoast-components`.
* Checkout it's companion branch in wordpress-seo. https://github.com/Yoast/wordpress-seo/pull/9635
* Install and build yoast-components: `yarn install && yarn start`.
* Start the wordpress-seo dev server: `yarn start`.
* In the `wp-config.php` file of your WordPress installation add `define( 'YOAST_FEATURE_SNIPPET_PREVIEW', true );` to enable the React SnippetPreview.
* Check if the new snippet preview is the same as the old snippet preview. Addressed for now is the following:

--

* Change mobile snippet header 16px Arial.
* Change mobile snippet meta description to 14px Arial.
* Change edit snippet button to a height of 33 px, to match size with the mobile/desktop switcher.
* Change edit snippet pencil color to grey dark.
* Change margin and padding around of button to match the original more closely.
* Change Titles of input fields to 16px Arial.
* Change margin on the progress bar to more closely match the original in terms of looks.
* Change top padding on the edit snippet section to more closely match the original.
* Change slug in the snippet preview to not go to newline, and become an ellipsis when too long. It should work that way based on what search results on google look like. 

--
Additionally fixed after CR:
* Changed line height for mobile description and for 
* Switcher white background on hover and focus. And with a think border when hovered, focused, or active.
* Matched the padding on input fields to those of the old editor.
* Gave the description input field a min-height. Anton decided a draggable input field wasn't essential, but that the height of the input field should be larger, to communicate more content was acceptable.
* Increased the space between seperate editor input fields, to more resemble the old snippet editor.
* The height of input fields is off by about two pixels. This is difficult to fix because draft.js creates those input fields. It has been discussed with Anton, and considered an acceptable discrepancy.
* The caret missing in the mobile desc is addressed in another pull: #513

Fixes https://github.com/Yoast/wordpress-seo/issues/9533
